torch/csrc/jit/script/compiler.h

```diff
@@ -123,14 +131,15 @@ struct BuiltinFunction : public SugaredValue {
 using Resolver = std::function<std::shared_ptr<SugaredValue>(const std::string& name)>;
 void defineMethodsInModule(
   Module & m,
-  const std::vector<Def>& definitions,
+  const std::vector<DefAndTypes>& definitions,
   const std::vector<Resolver>& resolvers, /* determines how we handle free variables in each definition*/
-  std::shared_ptr<SugaredValue> self /* if non-null, the first argument to each def, is bound to this value */
+  std::shared_ptr<SugaredValue> self, /* if non-null, the first argument to each def, is bound to this value */
+  bool pure_func=false /* If true, expect a single def which will become the 'forward' method on the module */
```

Comment:
Please remove this. It is poisoning this part of the code with knowledge of the `forward` convention. There is a lot of care taken in the compiler than lots of different concepts funnel through to one action: defining a list of methods on a module, and this change messes all of that up by re-introducing multiple things. That should be de-sugared well before we reach this point.